mantidproject/mantid
# Add minor ticks and minor gridlines settings

Report to: Rob Dalgliesh

**Description of work.**
This PR adds settings to the File->Settings window and the Axes tab in the figure options for showing minor ticks and minor gridlines on plots.

**To test:**
Go to File->Settings.
In the Plots tab there should now be check boxes for showing minor ticks and minor gridlines.
Check both "Show minor ticks" and "Show minor gridlines".
Make a plot and you should see minor ticks on the axes, clicking the grid button in the toolbar should show both major and minor gridlines.
Open the figure options and in the Axes tab you should be able to set whether the axes has minor ticks/gridlines on a per axes basis.
3D plots don't support minor ticks and gridlines so the settings don't apply to these and the options shouldn't be visible in the figure options.
If a plot has minor ticks, doing plot to script should include a command for showing the minor ticks.

Fixes #28546 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
